examples/connext_dds/dynamic_data_request_reply/cpp/PrimeNumberReplier.cxx

```diff
@@ -0,0 +1,328 @@
+            throw std::runtime_error("create_participant error");
+        }
+    }
+
+    ~PrimeNumberReplierExample()
+    {
+        /* Delete the participant and all contained entities */
+
+        DDS::ReturnCode_t retcode = participant->delete_contained_entities();
+        if (retcode != DDS::RETCODE_OK) {
+            fprintf(stderr, "delete_contained_entities error %d\n", retcode);
+        }
+
+        retcode = DDS::DomainParticipantFactory::get_instance()->
+            delete_participant(participant);
+        if (retcode != DDS::RETCODE_OK) {
+            fprintf(stderr, "delete_participant error %d\n", retcode);
+        }
+    }
+
+private:
+
+    void calculate_and_send_primes(
+        connext::Replier<DDS_DynamicData, DDS_DynamicData>& replier,
+        const DDS::SampleIdentity_t& request_sample_identity,
+        DDS_Long n,
+        DDS_Long primes_per_reply,
+        DDS_DynamicData *reply)
+        {
+
+        /* Set Status to REPLY_IN_PROGRESS */
+        reply->set_long("status", DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED,0);
+
+        /* Set the sequence: */
+        DDS_LongSeq primes_seq;
+        primes_seq.maximum(primes_per_reply);
+
+        /* prime[i] indicates if i is a prime number
+         * Initially, we assume all of them are prime
+         */
+        std::vector<bool> prime(n+1, true);
+
+        prime[0] = false;
+        prime[1] = false;
+
+        int m = (int) sqrt((float)n);
+
+        for (int i = 2; i <= m; i++) {
+            if (prime[i]) {
+                for (int k = i*i; k <= n; k+=i) {
+                    prime[k] = false;
+                }
+
+                /* Add a new element */
+                DDS::Long length = primes_seq.length();
+                primes_seq.length(length + 1);
+                primes_seq[length] = i;
+
+                if (length + 1 == primes_per_reply) {
+                    /* Send a reply now */
+                    reply->set_long_seq(
+                            "primes",
+                            DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED, 
+                            primes_seq);
+
+                    replier.send_reply(*reply, request_sample_identity);
+                    primes_seq.length(0);
+                }
+            }
+        }
+
+        /* Calculation is done. Send remaining prime numbers */
+        for (int i = m + 1; i <= n; i++) {
+            if (prime[i]) {
+
+                DDS::Long length = primes_seq.length();
+                primes_seq.length(length + 1);
+                primes_seq[length] = i;
+
+                if (length + 1 == primes_per_reply) {
+                    /* Send a reply now */
+                    reply->set_long_seq("primes",
+                        DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED, 
+                        primes_seq);
+                    replier.send_reply(*reply, request_sample_identity);
+                    primes_seq.length(0);
+                }
+            }
+        }
```

Comment:
I would divide this in 2 phases, one where we calculate the prime numbers (and store it into an array for example) and a second one where we send the sequences with the specific length